tests/integration/agents/test_graph.py

```diff
@@ -0,0 +1,234 @@
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(
+                    content="Create a hello world app with python and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app with python", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+    ],
+)
+def test_planner(
+    messages,
+    expected_answer,
+    general_query,
+    kyma_graph,
+    planner_correctness_metric,
+    answer_relevancy_metric,
+):
+    state = create_state(messages)
+    result = kyma_graph._invoke_planner(state)
+
+    test_case = LLMTestCase(
+        input=messages[-1].content,
+        actual_output=result.content,
+        expected_output=expected_answer,
+    )
+
+    if not general_query:
+        planner_correctness_metric.measure(test_case)
+
+        print(f"Score: {planner_correctness_metric.score}")
+        print(f"Reason: {planner_correctness_metric.reason}")
+
+        kyma_graph.plan_parser.parse(result.content)
+
+        assert_test(test_case, [planner_correctness_metric])
+    else:
+        assert_test(test_case, [answer_relevancy_metric])
+
+
+@pytest.mark.parametrize(
+    "messages, expected_answer",
+    [
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content="What is the capital of Germany?"),
+            ],
+            "Berlin",
+        ),
+        (
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content='Write "Hello, World!" code in Python'),
+            ],
+            'Here is a simple "Hello, World!" program in Python: `print("Hello, World!")`',
+        ),
+    ],
+)
+def test_common_node(messages, expected_answer, kyma_graph, answer_relevancy_metric):
```

Comment:
can you please add a comment that explains what the purpose of the test is, what is done etc.?